Mlem/API/Requests/Site/EditSite.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  EditSite.swift
+//  Mlem
+//
+//  Created by Jonathan de Jong on 12.06.2023.
+//
+
+import Foundation
+
+struct EditSiteRequest: APIPutRequest {
+
+    typealias Response = SiteResponse
+
+    let instanceURL: URL
+    let path = "site"
+    let body: Body
+
+    // lemmy_api_common::site::CreateSite
+    struct Body: Encodable {
+        let name: String?
+        let sidebar: String?
+        let description: String?
+        let icon: URL?
+        let banner: URL?
+        let enableDownvotes: Bool?
```

Comment:
you're about to hate me... but I just realised that all these `Body` structs are using `lowerCamelCase` for their fields, if you look in the previously created models I was using `snake_case` so that what we encode and send to the API is correct:
```swift
// lemmy_api_common::post::CreatePostLike
struct Body: Encodable {
    let auth: String
    let post_id: Int
    let score: Int
}
```

The encoder in the API is set to use the default key encoding strategy so will not convert these new ones, we could enable that (and update all the existing models) but given they're internal models only used within the request initialisers I'd gone the path of least resistance... and although likely not a consideration for the size of bodies we're encoding Apple state the following:

> Key encoding strategies other than JSONEncoder.KeyEncodingStrategy.useDefaultKeys may have a noticeable performance cost because those strategies may inspect and transform each key.

I just checked by changing the existing `CreatePostLike` call to use `postId` in it's body instead of `post_id` and the API doesn't like that and spits back:

> "Json deserialize error: missing field `post_id` at line 1 column 182"

Reply:
![image](https://github.com/buresdv/Mlem/assets/22740616/19fe4d3f-31dc-493f-908b-f4291dd1c474)

(Jk, it's no problem, thanks for catching this 💚)

Reply:
This is done now